program/tests/close_lookup_table_ix.rs

```diff
@@ -0,0 +1,229 @@
+#![cfg(feature = "test-sbf")]
+
+use {
+    address_lookup_table::instruction::close_lookup_table,
+    assert_matches::assert_matches,
+    common::{
+        add_lookup_table_account, assert_ix_error, new_address_lookup_table, setup_test_context,
+    },
+    solana_program_test::*,
+    solana_sdk::{
+        clock::Clock,
+        instruction::InstructionError,
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+        slot_hashes::SlotHashes,
+        sysvar::Sysvar,
+        transaction::Transaction,
+    },
+};
+
+mod common;
+
+#[tokio::test]
+async fn test_close_lookup_table() {
+    let mut context = setup_test_context().await;
+    context
+        .warp_to_slot(SlotHashes::size_of() as u64 + 1)
+        .unwrap();
+
+    let lookup_table_address = Pubkey::new_unique();
+    let authority_keypair = Keypair::new();
+    let initialized_table = {
+        let mut table = new_address_lookup_table(Some(authority_keypair.pubkey()), 0);
+        table.meta.deactivation_slot = 0;
+        table
+    };
+    add_lookup_table_account(&mut context, lookup_table_address, initialized_table).await;
+
+    let client = &mut context.banks_client;
+    let payer = &context.payer;
+    let recent_blockhash = context.last_blockhash;
+    let transaction = Transaction::new_signed_with_payer(
+        &[close_lookup_table(
+            lookup_table_address,
+            authority_keypair.pubkey(),
+            context.payer.pubkey(),
+        )],
+        Some(&payer.pubkey()),
+        &[payer, &authority_keypair],
+        recent_blockhash,
+    );
+
+    assert_matches!(client.process_transaction(transaction).await, Ok(()));
+    assert!(client
+        .get_account(lookup_table_address)
+        .await
+        .unwrap()
+        .is_none());
+}
+
+#[tokio::test]
+async fn test_close_lookup_table_not_deactivated() {
```

Comment:
To make this test a bit more interesting, can you warp exactly `512` slots to hit the limit for when it will still *fail*? The success test does `512 + 1`, so this will be perfect to make sure we're respecting the limits

Reply:
I think there's maybe some confusion between this test and `test_close_lookup_table_recently_deactivated`.

The idea behind this test - `test_close_lookup_table_not_deactivated` - is that no matter what slot we warp to, this is going to fail since the table has to be deactivated before it can be closed.

The other test - `test_close_lookup_table_recently_deactivated` - is doing that deactivation (by setting the `deactivation_slot`) and then attempting to close after slot warp.

So I think adding the edge casing you're describing here probably belongs in the other test.

Reply:
Ah gotcha, sorry for not understanding

Reply:
All good. I threw in some extra comments and what not to make sure it was more clear, for anyone else who might take a look or work on it!